includes/Customizer.php

```diff
@@ -0,0 +1,589 @@
+	public function validate_number_of_stories( $validity, $value ) {
+		$value = intval( $value );
+
+		if ( $value <= 0 ) {
+			$validity->add( 'invalid_number', __( 'The number of stories must be between 1 and 20.', 'web-stories' ) );
+		} elseif ( $value > 20 ) {
+			$validity->add( 'invalid_number', __( 'The number of stories must be between 1 and 20.', 'web-stories' ) );
+		}
+		return $validity;
+	}
+
+	/**
+	 * Validates the number of columns setting value.
+	 *
+	 * @param \WP_Error $validity WP_Error object.
+	 * @param int       $value Value to be validated.
+	 *
+	 * @return \WP_Error
+	 */
+	public function validate_number_of_columns( $validity, $value ) {
+		$value = intval( $value );
+
+		if ( $value <= 0 ) {
+			$validity->add( 'invalid_number', __( 'The number of stories must be between 1 and 4.', 'web-stories' ) );
+		} elseif ( $value > 5 ) {
+			$validity->add( 'invalid_number', __( 'The number of stories must be between 1 and 4.', 'web-stories' ) );
+		}
+		return $validity;
+	}
+
+	/**
+	 * Gets the stories theme support data.
+	 *
+	 * @return array An array of web story theme support data.
+	 */
+	public static function get_stories_theme_support() {
+
+		$theme_support = get_theme_support( 'web-story-options' );
+		$theme_support = ! empty( $theme_support[0] ) && is_array( $theme_support[0] ) ? $theme_support[0] : [];
+
+		$default_theme_support = [
+			'view-type'                 => [],
+			'view-type-default'         => 'circles',
+			'grid-columns-default'      => 2,
+			'title'                     => false,
+			'title-default'             => false,
+			'author'                    => false,
+			'author-default'            => false,
+			'date'                      => false,
+			'date-default'              => false,
+			'stories-archive-link'      => true,
+			'stories-archive-label'     => __( 'View all stories', 'web-stories' ),
+			'number-of-stories'         => 5,
+			'order'                     => [],
+			'order-default'             => 'oldest',
+			'show-story-poster-default' => true,
+		];
+
+		$theme_support = wp_parse_args( $theme_support, $default_theme_support );
+
+		$theme_support['view-type']            = is_array( $theme_support['view-type'] ) ? $theme_support['view-type'] : [];
+		$theme_support['order']                = is_array( $theme_support['order'] ) ? array_keys( $theme_support['order'] ) : [];
+		$theme_support['number-of-stories']    = is_numeric( $theme_support['number-of-stories'] ) ? $theme_support['number-of-stories'] : 5;
+		$theme_support['grid-columns-default'] = is_numeric( $theme_support['grid-columns-default'] ) ? $theme_support['grid-columns-default'] : 2;
+
+		return $theme_support;
+	}
+
+	/**
+	 * Renders web stories based on the customizer selected options.
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+	 *
+	 * @return string|void
+	 */
+	public static function render_stories() {
+		$options = get_option( 'story-options' );
```

Comment:
Options should be registered in the [Settings class](https://github.com/google/web-stories-wp/blob/main/includes/Settings.php).